### ISSUE TITLE
configure can't be told to build without openssl if it's detectable on the build system

### ISSUE THREAD

**Issue by pprindeville** [esnet/iperf#624]
# Context

* Version of iperf3:

3.2

* Hardware:

x86_64

* Operating system (and distribution, if any):

Cross building under Ubuntu 16.04 LTS for LEDE Head

# Bug Report

* Expected Behavior

You might have libopenssl on your system but still want to build iperf3 without iperf-auth (or any OpenSSL) encumbrance.

* Actual Behavior

If you specify `--disable-openssl` or `--without-openssl` that is ignored, and if you don't put anything, it autodetects openssl anyway, as shown by:

```
configure: WARNING: unrecognized options: --disable-openssl
...
checking whether compiling and linking against OpenSSL works... yes
...
/bin/bash ../libtool  --tag=CC   --mode=link x86_64-openwrt-linux-musl-gcc -g -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -iremap/home/philipp/bertram/lede/build_dir/target-x86_64_musl_powercode-bmu/iperf-3.2:iperf-3.2 -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -D_GNU_SOURCE  -Wall -g -L/home/philipp/bertram/lede/staging_dir/target-x86_64_musl_powercode-bmu/usr/lib -L/home/philipp/bertram/lede/staging_dir/target-x86_64_musl_powercode-bmu/lib -L/home/philipp/bertram/lede/staging_dir/toolchain-x86_64_gcc-5.4.0_musl/usr/lib -L/home/philipp/bertram/lede/staging_dir/toolchain-x86_64_gcc-5.4.0_musl/lib -znow -zrelro  -L/home/philipp/bertram/lede/staging_dir/target-x86_64_musl_powercode-bmu/usr/lib -o iperf3 iperf3-main.o libiperf.la -lssl -lcrypto 
```

* Steps to Reproduce

Install `openssl-devel` (or whatever) but try to build `iperf3` without it.

* Possible Solution

None so far.

**Comment by bmah888** [comment]
I see the problem you are reporting.  I had thought you'd be able to pass `--without-openssl` or `--with-openssl=no` and have it Do The Right Thing.  The autoconf infrastructure seems to handle this in a reasonable way, but the problem is that `config/ax_check_openssl.m4` seems to explicitly check for this case and blow up.  I think the solution will be in that file.

**Comment by bpetto35** [comment]
When building for android device, I cannot avoid the detection of the openssl on the build host.
I have tried to remove the line from m4 but I don't see why it keeps the ssl and crypto in libs.

            # no such luck; use some default ssldirs
            if ! $found; then
                ssldirs="/usr/local/ssl /usr/lib/ssl /usr/ssl /usr/pkg /usr/local /usr"
            fi

**Comment by pprindeville** [comment]
@bpetto35 I would recommend following up on this [thread](http://lists.gnu.org/archive/html/autoconf-archive-maintainers/2017-07/msg00016.html) if you want this to get fixed.

**Comment by bmah888** [comment]
@ralcini, FYI.

I think it's possible to fix this, because the detection logic for `--without-openssl` is already there.  It's just that autoconf doesn't do the right thing in that situation.

**Comment by ralcini** [comment]
Hi bmah888, 
I will make a pull request in the next few days to solve this problem.

**Comment by ralcini** [comment]
@pprindeville: please, can you test my fix?

**Comment by pprindeville** [comment]
@ralcini Works great as`--without-openssl`.

**Comment by bmah888** [comment]
Fix merged to master and the 3.2-STABLE branch.  Closing.
